Documentation/development-flow.md

```diff
@@ -0,0 +1,116 @@
+# Prerequisites
+1. GO installed
+2. Git client installed
+3. Github account
+
+# Initial Setup
+
+## Create a Fork
+
+From your browser navigate to <a href="https://github.com/rook/rook">http://github.com/rook/rook</a> and click the "Fork" button. 
+
+## Clone Your Fork
+
+Open a console window and do the following;
+```bash
+# Create the rook repo path
+mkdir -p $GOPATH/src/github.com/rook
+
+# Navigate to the local repo path and clone your fork
+cd $GOPATH/src/github.com/rook
+
+# Clone your fork, where <user> is your github account name
+git clone git@github.com:<user>/rook.git
+
+cd rook
+```
+
+## Add Upstream Remote
+First you will need to add the upstream remote to your local git:
+```bash
+# Add 'upstream' to the list of remotes
+git remote add upstream https://github.com/rook/rook.git
+
+# Verify the remote was added
+git remote -v
+```
+Now you should have at least `origin` and `upstream` remotes. You can also add other remotes to collaborate with other contributors.
+
+
+# Development
+To add a feature or to make a bug fix, you will need to create a branch in your fork and then submit a pull request (PR) from the branch.
+
+## Create a Branch
+From a console, create a new branch based on your fork and start working on it:
+```bash
+# Checkout the master branch - you want your new branch to come from master
+git checkout master
```

Comment:
you don't have to explicitly checkout master, you can create a branch that's directly based off `upstream/master` like so:
```
git checkout -b my-feature upstream/master
```